File: valid_palindrome.py
def isPalindrome(s: str) -> bool:
        start, end = 0 , len(s) - 1
        s = s.lower()
        while start < end:
            if not s[start].isalnum():
                start +=  1
                continue
            if not s[end].isalnum():
                end -= 1
                continue
            if s[start] != s[end]:
                return False
            start += 1
            end -= 1
        return True

File: cpp/valid_palindrome.cpp
#include <iostream>
#include <cctype>

using namespace::std;

class Solution {
public:
    bool isPalindrome(string s) {
        int st = 0, e = s.size() - 1;
        while (st < e) {
            if (!isalnum(s[st])) {
                st++;
                continue;
            }
            
            if (!isalnum(s[e])) {
                e--;
                continue;
            }
            
            if (tolower(s[st++]) != tolower(s[e--])) {
                return false;
            }
        }
        return true;
    }
};